packages/mainsail/source/transaction.service.ts

```diff
@@ -239,107 +156,23 @@ export class TransactionService extends Services.AbstractTransactionService {
-			transaction.senderPublicKey(senderPublicKey);
-		}
-
-		if (input.nonce) {
-			transaction.nonce(input.nonce);
-		} else {
-			const wallet = await this.clientService.wallet({ type: "address", value: address! });
-
-			transaction.nonce(wallet.nonce().plus(1).toFixed(0));
-		}
-
-		if (input.data && input.data.amount) {
-			transaction.amount(this.toSatoshi(input.data.amount).toString());
-		}
-
-		if (input.data && Array.isArray(input.data.payments)) {
-			for (const { amount, to } of input.data.payments) {
-				transaction.addPayment(to, BigNumber.make(this.toSatoshi(amount)).toString());
-			}
-		}
-
-		if (input.fee) {
-			transaction.fee(this.toSatoshi(input.fee).toString());
-		}
-
-		try {
-			if (input.data && input.data.expiration) {
-				transaction.expiration(input.data.expiration);
-			} else {
-				let estimatedExpiration: string | undefined;
-
-				if (
-					input.signatory.actsWithMultiSignature() ||
-					input.signatory.hasMultiSignature() ||
-					type === "multiSignature"
-				) {
-					estimatedExpiration = await this.estimateExpiration("211");
-				} else {
-					estimatedExpiration = await this.estimateExpiration("5");
-				}
-
-				if (estimatedExpiration) {
-					transaction.expiration(Number.parseInt(estimatedExpiration));
-				}
-			}
-		} catch {
-			// If we fail to set the expiration we'll still continue.
-		}
-
-		if (callback) {
-			callback({ data: input.data, transaction });
-		}
-
-		if (input.signatory.actsWithMultiSignature()) {
-			const serialized = await this.#app.resolve(Utils).toBytes(transaction.data);
-			const id = await this.#app.resolve(Utils).getId({ serialized } as MainsailContracts.Crypto.Transaction);
-
-			transaction.data.id = id.toString();
-			const transactionWithSignature = this.#multiSignatureSigner().sign(transaction, input.signatory.asset());
-
-			return this.dataTransferObjectService.signedTransaction(
-				transactionWithSignature.id!,
-				transactionWithSignature,
-			);
-		}
-
-		if (input.signatory.hasMultiSignature()) {
-			return await this.#addSignature(transaction, input.signatory.multiSignature()!, input.signatory);
-		}
+		return { address, publicKey };
+	}
 
-		if (type === "multiSignature") {
-			return await this.#addSignature(
-				transaction,
-				{
-					min: input.data.min,
-					publicKeys: input.data.publicKeys,
-				},
-				input.signatory,
-				senderPublicKey,
-			);
+	async #generateNonce(address?: string, input?: Services.TransactionInputs): Promise<string> {
+		if (input?.nonce) {
+			return input.nonce;
 		}
 
-		if (input.signatory.actsWithLedger()) {
-			transaction.data.signature = await this.#ledgerService.signTransaction(
-				input.signatory.signingKey(),
-				Transactions.Serializer.getBytes(transaction.data, {
-					excludeSignature: true,
-				}),
-			);
+		const wallet = await this.clientService.wallet({ type: "address", value: address! });
 
-			await this.#ledgerService.disconnect();
-		}
+		return wallet.nonce().plus(1).toFixed(0);
```

Comment:
mind that `nonce` in mainsail no longer needs to be incremented by 1 compared to what the API reports.

- core: API reports nonce 5 = use nonce 6 for the next tx
- mainsail: API reports nonce 5 = use nonce 5 for the next tx